# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1343786278 14400
#      Tue Jul 31 21:57:58 2012 -0400
# Node ID 000587f92082c74376b2b1e1ea9d52579e8d52ac
# Parent  b62b0b85369c2f1d09be7be657f909364c2796fe
rename src/DLD-FUNCTIONS directory to src/dldfcn

* src/dldfcn: Rename from src/DLD-FUNCTIONS.
* autogen.sh, src/Makefile.am, src/dldfcn/config-module.awk,
src/dldfcn/config-module.sh: Change all uses of DLD-FUNCTIONS to be
dldfcn.  Change all uses of DLD_FUNCTIONS to be DLDFCN.

diff --git a/autogen.sh b/autogen.sh
--- a/autogen.sh
+++ b/autogen.sh
@@ -25,19 +25,19 @@ done
 echo "generating source lists for liboctave/Makefile..."
 
 (cd liboctave; ./config-ops.sh)
 
 echo "generating doc/interpreter/images.mk..."
 
 (cd doc/interpreter; ./config-images.sh)
 
-echo "generating src/DLD-FUNCTIONS/module.mk..."
+echo "generating src/dldfcn/module.mk..."
 
-(cd src/DLD-FUNCTIONS; ./config-module.sh)
+(cd src/dldfcn; ./config-module.sh)
 
 echo "bootstrapping..."
 
 build-aux/bootstrap "$@"
 
 ## G77 is obsolete, but it is still the first option in the autoconf Fortran
 ## macros.  We should avoid it, because mixing old versions of g77 with modern
 ## gcc and g++ causes trouble.  The following will make it harder (but not
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -309,24 +309,24 @@ DIST_SRC = \
 
 noinst_LTLIBRARIES =
 
 include parse-tree/module.mk
 include octave-value/module.mk
 include operators/module.mk
 include template-inst/module.mk
 include corefcn/module.mk
-include DLD-FUNCTIONS/module.mk
+include dldfcn/module.mk
 
-$(srcdir)/DLD-FUNCTIONS/module.mk: $(srcdir)/DLD-FUNCTIONS/config-module.sh $(srcdir)/DLD-FUNCTIONS/config-module.awk $(srcdir)/DLD-FUNCTIONS/module-files
-	$(srcdir)/DLD-FUNCTIONS/config-module.sh $(top_srcdir)
+$(srcdir)/dldfcn/module.mk: $(srcdir)/dldfcn/config-module.sh $(srcdir)/dldfcn/config-module.awk $(srcdir)/dldfcn/module-files
+	$(srcdir)/dldfcn/config-module.sh $(top_srcdir)
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
-  OCT_FILES = $(DLD_FUNCTIONS_LIBS:.la=.oct)
-  OCT_STAMP_FILES = $(subst DLD-FUNCTIONS/,DLD-FUNCTIONS/$(am__leading_dot),$(DLD_FUNCTIONS_LIBS:.la=.oct-stamp))
+  OCT_FILES = $(DLDFCN_LIBS:.la=.oct)
+  OCT_STAMP_FILES = $(subst dldfcn/,dldfcn/$(am__leading_dot),$(DLDFCN_LIBS:.la=.oct-stamp))
   DLD_LIBOCTINTERP_LIBADD = liboctinterp.la
 else
   OCT_FILES =
   OCT_STAMP_FILES =
   DLD_LIBOCTINTERP_LIBADD =
 endif
 
 liboctinterp_la_SOURCES = \
@@ -382,27 +382,27 @@ octave_LDADD = \
 
 octave_LDFLAGS = \
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_LINK_OPTS)
 
 ## Section for defining and creating DEF_FILES
 SRC_DEF_FILES := $(shell $(srcdir)/find-defun-files.sh "$(srcdir)" $(DIST_SRC))
 
-DLD_FUNCTIONS_DEF_FILES = $(DLD_FUNCTIONS_SRC:.cc=.df)
+DLDFCN_DEF_FILES = $(DLDFCN_SRC:.cc=.df)
 
 ## builtins.cc depends on $(DEF_FILES), so DEF_FILES should only include
 ## .df files that correspond to sources included in liboctave.
 if AMCOND_ENABLE_DYNAMIC_LINKING
   DEF_FILES = $(SRC_DEF_FILES)
 else
-  DEF_FILES = $(SRC_DEF_FILES) $(DLD_FUNCTIONS_DEF_FILES)
+  DEF_FILES = $(SRC_DEF_FILES) $(DLDFCN_DEF_FILES)
 endif
 
-ALL_DEF_FILES = $(SRC_DEF_FILES) $(DLD_FUNCTIONS_DEF_FILES)
+ALL_DEF_FILES = $(SRC_DEF_FILES) $(DLDFCN_DEF_FILES)
 
 $(SRC_DEF_FILES): mkdefs Makefile
 
 $(DEF_FILES): $(OPT_HANDLERS) $(OPT_INC)
 
 DLL_CDEFS = @OCTINTERP_DLL_DEFS@
 DLL_CXXDEFS = @OCTINTERP_DLL_DEFS@
 
@@ -478,20 +478,20 @@ oct-errno.cc: oct-errno.in.cc Makefile
 $(OPT_HANDLERS) : %.cc : $(top_srcdir)/liboctave/%.in $(top_srcdir)/build-aux/mk-opts.pl
 	$(PERL) $(top_srcdir)/build-aux/mk-opts.pl --opt-handler-fcns $< > $@-t
 	mv $@-t $@
 
 $(OPT_INC) : %.h : %.in
 	$(MAKE) -C $(@D) $(@F)
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
-DLD_FUNCTIONS_PKG_ADD_FILE = DLD-FUNCTIONS/PKG_ADD
+DLDFCN_PKG_ADD_FILE = dldfcn/PKG_ADD
 
-DLD-FUNCTIONS/PKG_ADD: $(DLD_FUNCTIONS_DEF_FILES) mk-pkg-add
-	$(srcdir)/mk-pkg-add $(DLD_FUNCTIONS_DEF_FILES) > $@-t
+dldfcn/PKG_ADD: $(DLDFCN_DEF_FILES) mk-pkg-add
+	$(srcdir)/mk-pkg-add $(DLDFCN_DEF_FILES) > $@-t
 	mv $@-t $@
 endif
 
 lex.lo lex.o oct-parse.lo oct-parse.o: \
   AM_CXXFLAGS := $(filter-out -Wold-style-cast, $(AM_CXXFLAGS))
 
 __fltk_uigetfile__.lo __fltk_uigetfile__.o: \
   AM_CXXFLAGS := $(filter-out $(DLL_CXXDEFS), $(AM_CXXFLAGS) $(GRAPHICS_CFLAGS))
@@ -505,19 +505,19 @@ if AMCOND_BUILD_DOCS
 		cp $(srcdir)/DOCSTRINGS DOCSTRINGS; \
 		touch -r $(srcdir)/DOCSTRINGS DOCSTRINGS; \
 	fi
 	@echo "creating .DOCSTRINGS from .cc source files"
 	@$(PERL) $(srcdir)/gendoc.pl $(ALL_DEF_FILES) > $@
 	$(top_srcdir)/build-aux/move-if-change $@ DOCSTRINGS
 	touch $@
 
-all-local: $(OCT_STAMP_FILES) $(DLD_FUNCTIONS_PKG_ADD_FILE) .DOCSTRINGS
+all-local: $(OCT_STAMP_FILES) $(DLDFCN_PKG_ADD_FILE) .DOCSTRINGS
 else
-all-local: $(OCT_STAMP_FILES) $(DLD_FUNCTIONS_PKG_ADD_FILE)
+all-local: $(OCT_STAMP_FILES) $(DLDFCN_PKG_ADD_FILE)
 endif
 
 if AMCOND_BUILD_COMPILED_AUX_PROGRAMS
 octave-config.cc: octave-config.in.cc Makefile
 	@$(do_subst_default_vals)
 
 mkoctfile.cc: mkoctfile.in.cc Makefile
 	@$(do_subst_config_vals)
@@ -561,21 +561,21 @@ if ! AMCOND_BUILD_COMPILED_AUX_PROGRAMS
 	done
 endif
 
 .PHONY: make-version-links remove-version-links
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
 install-oct:
 	$(top_srcdir)/build-aux/mkinstalldirs $(DESTDIR)$(octfiledir)
-	if [ -n "`cat $(DLD_FUNCTIONS_PKG_ADD_FILE)`" ]; then \
-	  $(INSTALL_DATA) $(DLD_FUNCTIONS_PKG_ADD_FILE) $(DESTDIR)$(octfiledir)/PKG_ADD; \
+	if [ -n "`cat $(DLDFCN_PKG_ADD_FILE)`" ]; then \
+	  $(INSTALL_DATA) $(DLDFCN_PKG_ADD_FILE) $(DESTDIR)$(octfiledir)/PKG_ADD; \
 	fi
 	cd $(DESTDIR)$(octlibdir) && \
-	for ltlib in $(DLD_FUNCTIONS_LIBS); do \
+	for ltlib in $(DLDFCN_LIBS); do \
 	  f=`echo $$ltlib | $(SED) 's,.*/,,'`; \
 	  dl=`$(SED) -n -e "s/dlname='\([^']*\)'/\1/p" < $$f`; \
 	  if [ -n "$$dl" ]; then \
 	    $(INSTALL_PROGRAM) $$dl $(DESTDIR)$(octfiledir)/`echo $$f | $(SED) 's,^lib,,; s,\.la$$,.oct,'`; \
 	  else \
 	    echo "error: dlname is empty in $$ltlib!"; \
 	    exit 1; \
 	  fi; \
@@ -590,17 +590,17 @@ uninstall-oct:
 	  rm -f $(DESTDIR)$(octfiledir)/$$f; \
 	done
 	rm -f $(DESTDIR)$(octfiledir)/PKG_ADD
 endif
 .PHONY: install-oct uninstall-oct
 
 CLEANFILES = \
   $(bin_SCRIPTS) \
-  $(DLD_FUNCTIONS_PKG_ADD_FILE) \
+  $(DLDFCN_PKG_ADD_FILE) \
   graphics-props.cc \
   oct-parse.output
 
 DISTCLEANFILES = \
   .DOCSTRINGS \
   DOCSTRINGS \
   $(BUILT_NODISTFILES) \
   $(OCT_FILES) \
diff --git a/src/DLD-FUNCTIONS/__delaunayn__.cc b/src/dldfcn/__delaunayn__.cc
rename from src/DLD-FUNCTIONS/__delaunayn__.cc
rename to src/dldfcn/__delaunayn__.cc
diff --git a/src/DLD-FUNCTIONS/__dsearchn__.cc b/src/dldfcn/__dsearchn__.cc
rename from src/DLD-FUNCTIONS/__dsearchn__.cc
rename to src/dldfcn/__dsearchn__.cc
diff --git a/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc b/src/dldfcn/__fltk_uigetfile__.cc
rename from src/DLD-FUNCTIONS/__fltk_uigetfile__.cc
rename to src/dldfcn/__fltk_uigetfile__.cc
diff --git a/src/DLD-FUNCTIONS/__glpk__.cc b/src/dldfcn/__glpk__.cc
rename from src/DLD-FUNCTIONS/__glpk__.cc
rename to src/dldfcn/__glpk__.cc
diff --git a/src/DLD-FUNCTIONS/__init_fltk__.cc b/src/dldfcn/__init_fltk__.cc
rename from src/DLD-FUNCTIONS/__init_fltk__.cc
rename to src/dldfcn/__init_fltk__.cc
diff --git a/src/DLD-FUNCTIONS/__init_gnuplot__.cc b/src/dldfcn/__init_gnuplot__.cc
rename from src/DLD-FUNCTIONS/__init_gnuplot__.cc
rename to src/dldfcn/__init_gnuplot__.cc
diff --git a/src/DLD-FUNCTIONS/__magick_read__.cc b/src/dldfcn/__magick_read__.cc
rename from src/DLD-FUNCTIONS/__magick_read__.cc
rename to src/dldfcn/__magick_read__.cc
diff --git a/src/DLD-FUNCTIONS/__voronoi__.cc b/src/dldfcn/__voronoi__.cc
rename from src/DLD-FUNCTIONS/__voronoi__.cc
rename to src/dldfcn/__voronoi__.cc
diff --git a/src/DLD-FUNCTIONS/amd.cc b/src/dldfcn/amd.cc
rename from src/DLD-FUNCTIONS/amd.cc
rename to src/dldfcn/amd.cc
diff --git a/src/DLD-FUNCTIONS/ccolamd.cc b/src/dldfcn/ccolamd.cc
rename from src/DLD-FUNCTIONS/ccolamd.cc
rename to src/dldfcn/ccolamd.cc
diff --git a/src/DLD-FUNCTIONS/chol.cc b/src/dldfcn/chol.cc
rename from src/DLD-FUNCTIONS/chol.cc
rename to src/dldfcn/chol.cc
diff --git a/src/DLD-FUNCTIONS/colamd.cc b/src/dldfcn/colamd.cc
rename from src/DLD-FUNCTIONS/colamd.cc
rename to src/dldfcn/colamd.cc
diff --git a/src/DLD-FUNCTIONS/config-module.awk b/src/dldfcn/config-module.awk
rename from src/DLD-FUNCTIONS/config-module.awk
rename to src/dldfcn/config-module.awk
--- a/src/DLD-FUNCTIONS/config-module.awk
+++ b/src/dldfcn/config-module.awk
@@ -1,77 +1,77 @@
 BEGIN {
   FS = "|";
   nfiles = 0;
 
   print "## DO NOT EDIT -- generated from module-files by config-module.awk";
   print ""
   print "EXTRA_DIST += \\"
-  print "  DLD-FUNCTIONS/config-module.sh \\"
-  print "  DLD-FUNCTIONS/config-module.awk \\"
-  print "  DLD-FUNCTIONS/module-files \\"
-  print "  DLD-FUNCTIONS/oct-qhull.h"
+  print "  dldfcn/config-module.sh \\"
+  print "  dldfcn/config-module.awk \\"
+  print "  dldfcn/module-files \\"
+  print "  dldfcn/oct-qhull.h"
   print ""
 }
 /^#.*/ { next; }
 {
   nfiles++;
   files[nfiles] = $1;
   cppflags[nfiles] = $2;
   ldflags[nfiles] = $3;
   libraries[nfiles] = $4;
 } END {
   sep = " \\\n";
-  print "DLD_FUNCTIONS_SRC = \\";
+  print "DLDFCN_SRC = \\";
   for (i = 1; i <= nfiles; i++) {
     if (i == nfiles)
       sep = "\n";
-    printf ("  DLD-FUNCTIONS/%s%s", files[i], sep);
+    printf ("  dldfcn/%s%s", files[i], sep);
   }
   print "";
 
   sep = " \\\n";
-  print "DLD_FUNCTIONS_LIBS = $(DLD_FUNCTIONS_SRC:.cc=.la)";
+  print "DLDFCN_LIBS = $(DLDFCN_SRC:.cc=.la)";
   print "";
   print "if AMCOND_ENABLE_DYNAMIC_LINKING";
   print "";
-  print "octlib_LTLIBRARIES += $(DLD_FUNCTIONS_LIBS)";
+  print "octlib_LTLIBRARIES += $(DLDFCN_LIBS)";
   print "";
   print "## Use stamp files to avoid problems with checking timestamps";
   print "## of symbolic links";
   print "";
   for (i = 1; i <= nfiles; i++) {
     basename = files[i];
     sub (/\.cc$/, "", basename);
-    printf ("DLD-FUNCTIONS/$(am__leading_dot)%s.oct-stamp: DLD-FUNCTIONS/%s.la\n", basename, basename);
+    printf ("dldfcn/$(am__leading_dot)%s.oct-stamp: dldfcn/%s.la\n", basename, basename);
     print "\trm -f $(<:.la=.oct)";
     print "\tla=$(<F) && \\";
     print "\t  of=$(<F:.la=.oct) && \\";
-    print "\t  cd DLD-FUNCTIONS && \\";
+    print "\t  cd dldfcn && \\";
     print "\t  $(LN_S) .libs/`$(SED) -n -e \"s/dlname='\\([^']*\\)'/\\1/p\" < $$la` $$of && \\";
     print "\t  touch $(@F)";
     print "";
   }
   print "else";
   print "";
-  print "noinst_LTLIBRARIES += $(DLD_FUNCTIONS_LIBS)";
+  print "noinst_LTLIBRARIES += $(DLDFCN_LIBS)";
   print "";
   print "endif";
 
   for (i = 1; i <= nfiles; i++) {
     basename = files[i];
     sub (/\.cc$/, "", basename);
     print "";
-    printf ("DLD_FUNCTIONS_%s_la_SOURCES = DLD-FUNCTIONS/%s\n",
+    printf ("dldfcn_%s_la_SOURCES = dldfcn/%s\n",
 	    basename, files[i]);
     if (cppflags[i])
       {
-        printf ("DLD-FUNCTIONS/%s.df: CPPFLAGS += %s\n",
+        printf ("dldfcn/%s.df: CPPFLAGS += %s\n",
                 basename, cppflags[i]);
-        printf ("DLD_FUNCTIONS_%s_la_CPPFLAGS = $(AM_CPPFLAGS) %s\n",
+        printf ("dldfcn_%s_la_CPPFLAGS = $(AM_CPPFLAGS) %s\n",
                 basename, cppflags[i]);
       }
-    printf ("DLD_FUNCTIONS_%s_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED_LDFLAG) %s $(OCT_LINK_OPTS)\n",
+    printf ("dldfcn_%s_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED_LDFLAG) %s $(OCT_LINK_OPTS)\n",
             basename, ldflags[i]);
-    printf ("DLD_FUNCTIONS_%s_la_LIBADD = $(DLD_LIBOCTINTERP_LIBADD) ../liboctave/liboctave.la ../libcruft/libcruft.la %s $(OCT_LINK_DEPS)\n",
+    printf ("dldfcn_%s_la_LIBADD = $(DLD_LIBOCTINTERP_LIBADD) ../liboctave/liboctave.la ../libcruft/libcruft.la %s $(OCT_LINK_DEPS)\n",
             basename, libraries[i]);
   }
 }
diff --git a/src/DLD-FUNCTIONS/config-module.sh b/src/dldfcn/config-module.sh
rename from src/DLD-FUNCTIONS/config-module.sh
rename to src/dldfcn/config-module.sh
--- a/src/DLD-FUNCTIONS/config-module.sh
+++ b/src/dldfcn/config-module.sh
@@ -7,13 +7,13 @@ set -e
 if [ $# -eq 1 ]; then
   top_srcdir="$1";
 else
   top_srcdir="../.."
 fi
 
 move_if_change="$top_srcdir/build-aux/move-if-change"
 
-dld_dir=$top_srcdir/src/DLD-FUNCTIONS
+dld_dir=$top_srcdir/src/dldfcn
 
 $AWK -f $dld_dir/config-module.awk < $dld_dir/module-files > $dld_dir/module.mk-t
 
 $move_if_change $dld_dir/module.mk-t $dld_dir/module.mk
diff --git a/src/DLD-FUNCTIONS/convhulln.cc b/src/dldfcn/convhulln.cc
rename from src/DLD-FUNCTIONS/convhulln.cc
rename to src/dldfcn/convhulln.cc
diff --git a/src/DLD-FUNCTIONS/dmperm.cc b/src/dldfcn/dmperm.cc
rename from src/DLD-FUNCTIONS/dmperm.cc
rename to src/dldfcn/dmperm.cc
diff --git a/src/DLD-FUNCTIONS/eigs.cc b/src/dldfcn/eigs.cc
rename from src/DLD-FUNCTIONS/eigs.cc
rename to src/dldfcn/eigs.cc
diff --git a/src/DLD-FUNCTIONS/fftw.cc b/src/dldfcn/fftw.cc
rename from src/DLD-FUNCTIONS/fftw.cc
rename to src/dldfcn/fftw.cc
diff --git a/src/DLD-FUNCTIONS/module-files b/src/dldfcn/module-files
rename from src/DLD-FUNCTIONS/module-files
rename to src/dldfcn/module-files
diff --git a/src/DLD-FUNCTIONS/oct-qhull.h b/src/dldfcn/oct-qhull.h
rename from src/DLD-FUNCTIONS/oct-qhull.h
rename to src/dldfcn/oct-qhull.h
diff --git a/src/DLD-FUNCTIONS/qr.cc b/src/dldfcn/qr.cc
rename from src/DLD-FUNCTIONS/qr.cc
rename to src/dldfcn/qr.cc
diff --git a/src/DLD-FUNCTIONS/symbfact.cc b/src/dldfcn/symbfact.cc
rename from src/DLD-FUNCTIONS/symbfact.cc
rename to src/dldfcn/symbfact.cc
diff --git a/src/DLD-FUNCTIONS/symrcm.cc b/src/dldfcn/symrcm.cc
rename from src/DLD-FUNCTIONS/symrcm.cc
rename to src/dldfcn/symrcm.cc
diff --git a/src/DLD-FUNCTIONS/tsearch.cc b/src/dldfcn/tsearch.cc
rename from src/DLD-FUNCTIONS/tsearch.cc
rename to src/dldfcn/tsearch.cc
diff --git a/src/DLD-FUNCTIONS/urlwrite.cc b/src/dldfcn/urlwrite.cc
rename from src/DLD-FUNCTIONS/urlwrite.cc
rename to src/dldfcn/urlwrite.cc
diff --git a/src/link-deps.mk b/src/link-deps.mk
--- a/src/link-deps.mk
+++ b/src/link-deps.mk
@@ -1,14 +1,14 @@
 include ../liboctave/link-deps.mk
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
   LIBOCTINTERP_LINK_DEPS =
 else
-  LIBOCTINTERP_LINK_DEPS = $(DLD_FUNCTIONS_LIBS)
+  LIBOCTINTERP_LINK_DEPS = $(DLDFCN_LIBS)
 endif
 
 LIBOCTINTERP_LINK_DEPS += \
   $(GRAPHICS_LIBS) \
   $(FT2_LIBS) \
   $(HDF5_LIBS) \
   $(Z_LIBS) \
   $(FFTW_XLIBS) \
